modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -500,6 +545,189 @@ public int getZoneId(String name) {
+                        immediateScaleUp0 = zone.dataNodesAutoAdjustScaleUp() == 0;
+                        immediateScaleDown0 = zone.dataNodesAutoAdjustScaleDown() == 0;
+
+                        break;
+                    }
+                }
+            }
+
+            immediateScaleUp.set(immediateScaleUp0);
+            immediateScaleDown.set(immediateScaleDown0);
+        });
+
+        CompletableFuture<Void> topVerScaleUpFut = new CompletableFuture<>();
+
+        topVerFut.thenAccept(ignored -> {
+            synchronized (dataNodesMutex) {
+                ZoneState zoneState = zonesState.get(zoneId);
+
+                if (zoneState == null) {
+                    throw new DistributionZoneNotFoundException(zoneId);
+                }
+
+                CompletableFuture<Void> topVerScaleUpFut0 = null;
+
+                if (immediateScaleUp.get()) {
+                    Map.Entry<Long, Long> scaleUpRevisionEntry = topVerAndScaleUpRevision.ceilingEntry(topVer);
+
+                    Long scaleUpRevision = null;
+
+                    if (scaleUpRevisionEntry != null) {
+                        scaleUpRevision = scaleUpRevisionEntry.getValue();
+                    }
+
+                    if (scaleUpRevision != null && zoneState.scaleUpRevision() < scaleUpRevision) {
+                        Map.Entry<Long, CompletableFuture<Void>> ceilingEntry =
+                                zoneState.revisionScaleUpFutures().ceilingEntry(scaleUpRevision);
+
+                        if (ceilingEntry != null) {
+                            topVerScaleUpFut0 = ceilingEntry.getValue();
+                        }
+
+                        if (topVerScaleUpFut0 == null) {
+                            topVerScaleUpFut0 = new CompletableFuture<>();
+
+                            zoneState.revisionScaleUpFutures().put(scaleUpRevision, topVerScaleUpFut0);
+                        }
+
+                        topVerScaleUpFut0.handle((ignored0, e) -> {
+                            if (e == null) {
+                                topVerScaleUpFut.complete(null);
+                            } else {
+                                topVerScaleUpFut.completeExceptionally(e);
+                            }
+
+                            return null;
+                        });
+                    } else {
+                        topVerScaleUpFut.complete(null);
+                    }
+                } else {
+                    topVerScaleUpFut.complete(null);
+                }
+            }
+        });
+
+        CompletableFuture<Void> topVerScaleDownFut = new CompletableFuture<>();
+
+        topVerFut.thenAccept(ignored -> {
+            CompletableFuture<Void> topVerScaleDownFut0 = null;
+
+            synchronized (dataNodesMutex) {
+                ZoneState zoneState = zonesState.get(zoneId);
+
+                if (zoneState == null) {
+                    throw new DistributionZoneNotFoundException(zoneId);
```

Comment:
How it's possible to specify incorrect zoneId. It might be an assertion, but not DistributionZoneNotFoundException because we have zone existence validation on table create/altering etc.

Reply:
Fixed. As we discussed we will use this method in the configuration closure so we don't need to throw this exception.